---
title: "RWT-46_PRC2_1x_0.1_combined_full proteome"
author: "Robert Warneford-Thomson"
date: "7/18/2016"
output: html_document
---

---
title: "RWT-46 PRC2 IP_RBRID 1x data"
author: "Robert Warneford-Thomson"
date: "6/24/2016"
output: html_document
---

```{r Initialization, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message=FALSE)
require(seqinr)
require(GenomicRanges)
require(gplots)
require(gdata)
require(parallel)

Experiment = "RWT46"
Descriptor = "Combined1x0.1xdata_fulldatabase"
Date = "18Jul2016" 
if (!any(grepl(Experiment, list.dirs("~/analyses/")))) {dir.create(path = paste0("~/analyses/", Experiment))}
```

## Import RBR-ID data
```{r}
library(gdata)
installXLSXsupport()
Prot_table <-read.xls(xls = "Robert test 1 with mouse proteome.xlsx", sheet = 1)
```

# Load Functions
```{r}
NormRBR <-function(Data = Prot_table, minusCols = c(9:16), plusCols = c(17:22), pvalcutoff = 0.1, depletionfactor = 0.95)
  
  ## Function to normalize and analyze RBR peptide intensities. MinusCols and plusCols are numeric vectors defining, respectively, columns containing -4sU and +4sU replicates.
  
{
MinusSamples <- names(Data[minusCols])
PlusSamples <- names(Data[plusCols])
RawSamples <- names(Data[c(minusCols, plusCols)])

# Rename data columns
colnames(Data)[grepl("contaminant", colnames(Data))] <- "contaminant"
colnames(Data)[grepl("Proteins", colnames(Data))] <- "uniprotID"
colnames(Data)[grepl("Protein.names", colnames(Data))] <- "name"
colnames(Data)[grepl("Gene.names", colnames(Data))] <- "symbol"
colnames(Data)[grepl("Unique..Groups.", colnames(Data))] <- "unique"

# Convert Contaminants to logical vector
Data$contaminant <- as.character(Data$contaminant)
Data$contaminant[Data$contaminant == "+"] <- T
Data$contaminant[Data$contaminant != "TRUE"] <- F
Data$contaminant<-as.logical(Data$contaminant)
Data$uniprotID <- as.character(Data$uniprotID)
Data$symbol <- as.character(Data$symbol)
Data$name <- as.character(Data$name)

# Convert Unique to logical vector
Data$unique <- as.character(Data$unique)
Data$unique[Data$unique =="yes"]<-T
Data$unique[Data$unique !="TRUE"]<-F
Data$unique<-as.logical(Data$unique)

# Assigns unique peptide ID to each peptide 
Data[,RawSamples] <- as.numeric(unlist(Data[,c(minusCols, plusCols)]))
Data <- cbind(Pept_ID = c(1:nrow(Data)), Data)

# Normalize peptide intensity by Sum of total intensity (removing contaminants) in each replicate AKA set Euclidean length = 1

Norm_data <- sweep(Data[,RawSamples], 2, colSums(Data[!Data$contaminant, RawSamples]), FUN = "/")
names(Norm_data) <- paste0(RawSamples,"_Norm")
Data <- cbind(Data, Norm_data)

# Update sample names
MinusSamples_Norm <- paste0(MinusSamples, "_Norm")
PlusSamples_Norm <- paste0(PlusSamples, "_Norm")

# Compute averages of biological replicates
Data$minus_mean <- rowMeans(Data[,MinusSamples_Norm])
Data$plus_mean <- rowMeans(Data[,PlusSamples_Norm])

# Calculate +4su/-4sU log2-fold change
Data$log2fold <- log2(Data$plus_mean/Data$minus_mean)
Data$log2fold[is.infinite(Data$log2fold) | is.nan(Data$log2fold)] <- NA

# Calculate p values from two-sided t-test
Data$p.value <- apply(Data[,c(PlusSamples_Norm, MinusSamples_Norm)], 1, function(x) 
{ if (sum(!is.na(x[PlusSamples_Norm])) < 1 | sum(!is.na(x[MinusSamples_Norm])) < 1)  {return(1)}
  
  else { t.test( x[PlusSamples_Norm], x[MinusSamples_Norm], paired = F, alternative = "two.sided")$p.value}})

# Cut-offs
# Remove data with p value > 0.1
Data <- Data[Data$p.value < pvalcutoff & !is.na(Data$p.value),]

# Include data with depletion greater than 5%
Data <- Data[Data$plus_mean/Data$minus_mean < depletionfactor ,]

# Calculate RBR-ID score
Data$RBRscore <- log(Data$p.value)*Data$log2fold

return(Data)}

subsetID <- function(names = c("P19909"), category = col, data = df)
  # Function to extract rows that match desired value of string variable, e.g. specific uniprot IDs.
  # names is vector of strings listing desired values to subset, category gives column to search for desired names
{ match.list <-lapply(names, function(x) {grep(pattern = paste0(".*", x,".*$"), data[,category], value = FALSE)})
  match.rows <- unlist(match.list)
  return(match.rows)}


calculateCoverage.apply<-function(range,peps)
{
  hits<-subsetByOverlaps(peps,range)
  covered.regions<-reduce(hits)
  sum(width(covered.regions))/end(range)
}

residueScorePlot<-function(range,peps,smooth=T, show.coverage=T, no.negs=F,span=0.0001, return.values=F)
  ## Draws score plot, smoothing and white space on NA peptides can be controlled by options
  ## LAST UPDATED 4.1.2016
{
  score<-rep(0,end(range))
  hits<-subsetByOverlaps(peps,range)
  for(i in 1:length(hits)){
    pep.pos<-start(hits[i]):end(hits[i])
    score[pep.pos]=score[pep.pos]+hits$score[i]
  }
  gaps<-setdiff(range,reduce(hits))
  for(i in 1:length(gaps)){
    gap.pos<-start(gaps[i]):end(gaps[i])
    score[gap.pos]<-NA
  }
  score.mod<-score
  score.mod[is.na(score.mod)]<-0
  if(no.negs) score.mod[score.mod<0]<-0
  if(smooth){
    values<-supsmu(1:length(score.mod),score.mod,span=span)
  } else {
    values<-(list(x=1:length(score.mod),y=score.mod))
  }
  plot(values, type="l", main=paste0(range$symbol, " (",seqnames(range),")"), xlab="residue #", ylab="RBR-ID score")
  if(show.coverage) points(c(1:length(score))[is.na(score)],rep(0,sum(is.na(score))), pch=16, col="white", cex=500/end(range))
  if(return.values) return(values)
}

residueScore<-function(range,peps,na=T)
  ## Returns a vector with score per residue
{
  score<-rep(0,end(range))
  hits<-subsetByOverlaps(peps,range)
  for(i in 1:length(hits)){
    pep.pos<-start(hits[i]):end(hits[i])
    score[pep.pos]=score[pep.pos]+hits$score[i]
  }
  if(na){
    gaps<-setdiff(range,reduce(hits))
    for(i in 1:length(gaps)){
      gap.pos<-start(gaps[i]):end(gaps[i])
      score[gap.pos]<-NA
    }
  }
  score
}

binMean <- function(data, bins)
  ## Calculate mean in a bin
{
  total<-length(data)
  internal.bins<-bins-2
  binSize<-floor(total/internal.bins)
  extra<-floor((total-binSize*internal.bins)/2)
  binned<-rep(0,bins)
  binned[1]<-mean(data[1:extra],na.rm=T)
  for(i in 1:(internal.bins)){
    binStart<-(i*binSize-binSize+1)+extra
    binEnd<-(i*binSize)+extra
    binned[i+1]<-mean(data[binStart:binEnd], na.rm=T)
    # print(paste(binStart,binEnd,binned[i+1], sep="  "))
  }
  binned[bins]<-mean(data[binEnd:total], na.rm=T)
  return(binned)
}
```


## Sanitize Sample names + Add description
```{r}
samples <-c("no4su_01","no4su_02","no4su0.1x_01", "no4su0.1x_02","4su_01","4su_02","4su0.1x_01","4su_0.1x_02")
colnames(Prot_table)[8:15] <- samples

Prot_table_Combined <- NormRBR(Data = Prot_table, minusCols = c(8, 9), plusCols = c(12, 13))

# Correct errant uniprot IDs 
Prot_table_Combined$uniprotID[Prot_table_Combined$symbol == "Ezh2"] <- "Q61188"
```


## Files
``` {r}

# Create table of protein lengths from Uniprot mouse proteome
Proteome <-read.fasta(file = "Mouse_proteome_15Jul2016.fasta", seqtype = "AA", as.string = T)

uniprotID <- sapply(names(Proteome), function(x){strsplit(x, split = "[|]")[[c(1, 2)]]}, USE.NAMES = F)
len <- unname(sapply(Proteome, function(x){nchar(x)[1]}))
Lengths <- data.frame(uniprotID = uniprotID, Length = len, stringsAsFactors = F)

# loads annotation from peptideAnnotation table

anno <- Prot_table_Combined
prot.anno<-anno[!anno$contaminant,c("uniprotID","symbol","name")]
prot.anno<-prot.anno[order(prot.anno$name, decreasing=T),]
prot.anno<-prot.anno[!duplicated(prot.anno$uniprotID),]
prot.anno<-prot.anno[order(prot.anno$uniprotID),]
prot.anno<-merge(prot.anno, Lengths, by="uniprotID")

# convert to grange
prot.gr<- GRanges(seqnames=Rle(prot.anno$uniprotID), ranges=IRanges(start=1, end=prot.anno$Length))
prot.gr$symbol <-prot.anno$symbol

# prepare peptide grange
pep.gr<-GRanges(seqnames=Rle(anno$uniprotID),ranges=IRanges(start=anno$Start.position,end=anno$End.position))
pep.gr$pepID<-anno$pept_ID; pep.gr$unique<-anno$unique; pep.gr$contaminant<-anno$contaminant; pep.gr$sequence<-anno$Sequence
```


# Plot score for PRC2 subunits
  Additionally I am outputting an attribute file that can be rendered using Chimera onto PDB structures
``` {r scoreplot}

# calculate coverage per protein
{
  rm(list=c("end","start")) # this is because the worker nodes behave strangely otherwise
  cl<-makeCluster(15)
  clusterExport(cl, list("subsetByOverlaps","reduce","width","end")) # must pass non-conventional functions to worker nodes
  coverage<-parSapply(cl=cl,prot.gr,calculateCoverage.apply, peps=pep.gr) # this takes about 1'15" with 30 nodes
  stopCluster(cl)
  prot.gr$coverage<-coverage*100
  saveRDS(prot.gr,"intermediate_files/prot.gr.coverage.160715.rds")
  prot.gr<-readRDS("intermediate_files/prot.gr.coverage.160715.rds")
}

# score plots
{
  ## calculate score
  pep.gr$score<-anno$RBRscore #note that you have to use the $ sign to specify the column with 5th replicates (regex)
  saveRDS(pep.gr, "intermediate_files/pep.gr.score312.160715.rds")
  
  ## plot score
 PRC2_names = c("Ezh2", "Eed", "Suz12", "Aebp2", "Jarid2", "Mtf2")
 

lapply(PRC2_names, function(candidate, PDB = "5IJ7") {
  
  if (!any(grepl(candidate, prot.gr$symbol))) {return(paste0("Candidate ", candidate, " is not identified"))} 
          else {
                pos<-which(prot.gr$symbol==candidate)
                t<-residueScorePlot(prot.gr[pos,],pep.gr,smooth=T, no.negs=T, return.values=T, span=0.001)
                
                a<-as.data.frame(pep.gr[seqnames(pep.gr)==as.character(seqnames(prot.gr)[pos]),])
                a<-a[order(a$score),]
                
                # to make score file that gets inserted in PDB for coloring
                t$label<-paste0(":",t$x)
                RBRscoretable<- cbind(t$label, round(t$y*10)/10)
                RBRscoretable
                fileName <- paste0("~/analyses/", Experiment, "/", Descriptor, Date, candidate, "scoreSmooth_", PDB, ".txt")
                write.table(RBRscoretable,sep="\t",quote=F, file= fileName, row.names = F, col.names = F) 
                
                header <- c(paste0("#  PDB entry ", PDB, " - " ,candidate, " RBR-ID RNA binding prediction score"),
                                  "#  (4sU crosslinked peptide mass spectrometry depletion)", 
                                  "#  Method credit to Chongsheng He & Roberto Bonasio",
                                  paste0("#", "  Experiment: ", Experiment),
                                  paste0("#", "  Details: ", Descriptor),
                                  "#  Use this file to assign the attribute to crystal structures in in Chimera with the",
                                  "#  Define Attribute tool or the command defattr.",
                                  paste0("#  Make sure to specify attribute only to ", candidate, " chain"),
                                  "#",
                                  "attribute: rbrscore",
                                  "match mode: 1-to-1",
                                  "recipient: residues")
                
                fConn <- file(fileName)
                Lines <- readLines(fConn) 
                writeLines(c(header, Lines), sep = "\n" , con = fConn)
                close(fConn)
          }})
}



```

